lib/pushmi_pullyu/cli.rb

```diff
@@ -9,19 +12,18 @@ class PushmiPullyu::CLI
 
   COMMANDS = ['start', 'stop', 'restart', 'reload', 'run', 'zap', 'status'].freeze
 
-  attr_accessor :config
+  def initialize; end
 
-  def initialize
-    self.config = PushmiPullyu::Config.new
-  end
+  def parse(argv = ARGV)
+    opts = parse_options(argv)
+    opts[:daemonize] = true if COMMANDS.include? argv[0]
+    opts = parse_config(opts[:config_file]).merge(opts) if opts[:config_file]
 
-  def parse(args = ARGV)
-    parse_options(args)
-    parse_commands(args)
+    options.merge!(opts)
```

Comment:
This is the merge call that could be encapsulated inside the `options=`, I think?